orc8r/cloud/go/services/configurator/storage/sql_entity_load_helpers.go

```diff
@@ -73,25 +79,29 @@ func (store *sqlConfiguratorStorage) getLoadEntitiesSelectBuilder(networkID stri
 				sq.Eq{fmt.Sprintf("ent.%s", entTypeCol): id.Type},
 			})
 		})
-		selectBuilder = selectBuilder.Where(orClause)
+		return selectBuilder.Where(orClause), nil
+	}
+	if filter.PhysicalID != nil {
+		return selectBuilder.Where(sq.Eq{fmt.Sprintf("ent.%s", entPidCol): filter.PhysicalID.Value}), nil
+	} else if filter.GraphID != nil {
+		return selectBuilder.Where(sq.Eq{fmt.Sprintf("ent.%s", entGidCol): filter.GraphID.Value}), nil
 	} else {
-		if filter.PhysicalID != nil {
-			selectBuilder = selectBuilder.Where(sq.Eq{fmt.Sprintf("ent.%s", entPidCol): filter.PhysicalID.Value})
-		} else if filter.GraphID != nil {
-			selectBuilder = selectBuilder.Where(sq.Eq{fmt.Sprintf("ent.%s", entGidCol): filter.GraphID.Value})
-		} else {
-			andClause := sq.And{sq.Eq{fmt.Sprintf("ent.%s", entNidCol): networkID}}
-			if filter.KeyFilter != nil {
-				andClause = append(andClause, sq.Eq{fmt.Sprintf("ent.%s", entKeyCol): filter.KeyFilter.Value})
-			}
-			if filter.TypeFilter != nil {
-				andClause = append(andClause, sq.Eq{fmt.Sprintf("ent.%s", entTypeCol): filter.TypeFilter.Value})
-			}
-			selectBuilder = selectBuilder.Where(andClause)
+		andClause := sq.And{sq.Eq{fmt.Sprintf("ent.%s", entNidCol): networkID}}
+		if filter.KeyFilter != nil {
+			andClause = append(andClause, sq.Eq{fmt.Sprintf("ent.%s", entKeyCol): filter.KeyFilter.Value})
+		}
+		if filter.TypeFilter != nil {
+			andClause = append(andClause, sq.Eq{fmt.Sprintf("ent.%s", entTypeCol): filter.TypeFilter.Value})
+		}
+		if criteria.PageSize > 0 {
+			andClause = append(andClause, sq.Gt{fmt.Sprintf("ent.%s", entKeyCol): criteria.PageToken})
+		}
+		selectBuilder = selectBuilder.Where(andClause)
+		if criteria.PageSize > 0 {
```

Comment:
Always set query limit. Use criteria.GetPageSize() described in other comment. Let's make the default max like 10 or 15k, with a note in the service config that thing will start to break much above 15k